app/controllers/admin/users_controller.rb

```diff
@@ -100,7 +101,25 @@ def remove_role
   private
 
   def user_params
-    params.require(:user).permit(:name, :organization_id, :email, :password, :password_confirmation)
+    user_params = params.require(:user).permit(:name, :organization_id, :email, :password, :password_confirmation)
+    role_id = updated_role_id
+
+    if role_id
+      user_params[:organization_role_join_attributes] = { role_id: role_id }
+    else
+      redirect_back(fallback_location: edit_admin_user_path, error: "Please select an organization")
```

Comment:
Not sure if the problem here is that the user didn't select an organization. It seems more like they _did_ (`organization_id` might not be null) but the role couldn't be found for some reason. Maybe make this error a bit more verbose and explain that? Or split it out into two error conditions, one where organization\_id is null and one where the role couldn't be found?

Reply:
Yeah okay, it makes sense to consider both cases. I assumed if the organization_id was provided the roles would be predefined and available in the database, but thanks for the pointer, I will handle both conditions and provide clear flash messages.